bundles/org.openhab.binding.nuvo/src/main/java/org/openhab/binding/nuvo/internal/handler/NuvoHandler.java

```diff
@@ -0,0 +1,776 @@
+                    closeConnection();
+                } else {
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, this.versionString);
+                }
+            }
+        }, 1, RECON_POLLING_INTERVAL, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Cancel the reconnection job
+     */
+    private void cancelReconnectJob() {
+        ScheduledFuture<?> reconnectJob = this.reconnectJob;
+        if (reconnectJob != null) {
+            reconnectJob.cancel(true);
+            this.reconnectJob = null;
+        }
+    }
+
+    /**
+     * Schedule the polling job
+     */
+    private void schedulePollingJob() {
+        logger.debug("Schedule polling job");
+        cancelPollingJob();
+
+        // when the Nuvo amp is off, this will keep the connection (esp Serial over IP) alive and detect if the
+        // connection goes down
+        pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+            if (connector.isConnected()) {
+                logger.debug("Polling the component for updated status...");
+
+                synchronized (sequenceLock) {
+                    try {
+                        connector.sendCommand(NuvoCommand.GET_CONTROLLER_VERSION);
+                    } catch (NuvoException e) {
+                        logger.debug("Polling error: {}", e.getMessage());
+                    }
+
+                    // if the last event received was more than 1.25 intervals ago,
+                    // the component is not responding even though the connection is still good
+                    if ((System.currentTimeMillis() - lastEventReceived) > (POLLING_INTERVAL * 1.25 * 1000)) {
+                        logger.debug("Component not responding to status requests");
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "Component not responding to status requests");
+                        closeConnection();
+                        scheduleReconnectJob();
+                    }
+                }
+            }
+        }, INITIAL_POLLING_DELAY, POLLING_INTERVAL, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Cancel the polling job
+     */
+    private void cancelPollingJob() {
+        ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+    }
+
+    /**
+     * Schedule the clock sync job
+     */
+    private void scheduleClockSyncJob() {
+        logger.debug("Schedule clock sync job");
+        cancelClockSyncJob();
+        clockSyncJob = scheduler.scheduleWithFixedDelay(() -> {
+            if (this.isGConcerto) {
+                try {
+                    SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy,MM,dd,HH,mm");
+                    connector.sendCommand(NuvoCommand.CFGTIME.getValue() + simpleDateFormat.format(new Date()));
```

Comment:
Did you try this with different system timezones and configuring the timezone in PaperUI? I saw an example retrieving a zoned datetime service or so via `@Reference` in an openHAB binding, but I can't find it again.

Reply:
I did not. It was just intended to use whatever the system time is regardless of time zone. I think this will be fine since most users set the host system to their local time.